src/csharp/Microsoft.Spark/Interop/Ipc/JsonSerDe.cs

```diff
@@ -2,70 +2,154 @@
 // The .NET Foundation licenses this file to you under the MIT license.
 // See the LICENSE file in the project root for more information.
 
-using System.Linq;
-using Newtonsoft.Json.Linq;
+using System;
+using System.Buffers;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Text;
+using System.Text.Json;
 
 namespace Microsoft.Spark.Interop.Ipc
 {
     /// <summary>
-    /// Json.NET Serialization/Deserialization helper class.
+    /// System.Text.Json Serialization/Deserialization helper class.
     /// </summary>
     internal static class JsonSerDe
     {
+
         /// Note: Scala side uses JSortedObject when parsing JSON, so the properties
-        /// in JObject need to be sorted.
+        /// in JsonElement need to be sorted.
         /// <summary>
         /// Extension method to sort items in a JSON object by keys.
         /// </summary>
-        /// <param name="jObject"></param>
+        /// <param name="jsonElement"></param>
+        /// <returns></returns>
+        public static string SortProperties(this JsonElement jsonElement)
```

Comment:
nit: Since this method is public (granted on an internal class), it might be worth noting as a remark in the xml docs that this uses recursion and shouldn't be used with untrusted/unbounded input data. 

Mainly to avoid folks calling `SortProperties` on arbitrary `JsonElement` in a different context (as is) than what it is currently used for (which is a closed set/bounded input).